### PR TITLE
Test getters

### PR DESCRIPTION
**Pull request by prithvi081099**

### CODE REVIEW

File: space-cli/cmd/modules/auth/getters_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		name           string
+		args           args
+		schemaMockArgs []mockArgs
+		url            map[string]interface{}
+		want           []*model.SpecObject
+		wantErr        bool
+	}{
+		// TODO: Add test cases.
+		{
+			name: "test-1",
```
Comment (sharadregoti) [inline space-cli/cmd/modules/auth/getters_test.go:40]: Give an appropiate test name

File: space-cli/cmd/modules/auth/getters_test.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+			args: args{
+				project:     "myproject",
+				commandName: "auth-providers",
+				params:      map[string]string{},
+			},
+			schemaMockArgs: []mockArgs{
```
Comment (sharadregoti) [inline space-cli/cmd/modules/auth/getters_test.go:46]:
```suggestion
			transportMockArgs: []mockArgs{
```

File: space-cli/cmd/modules/auth/getters_test.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		method         string
+		args           []interface{}
+		paramsReturned []interface{}
+	}
+	type args struct {
+		project     string
+		commandName string
+		params      map[string]string
+	}
+	type provider struct {
```
Comment (sharadregoti) [inline space-cli/cmd/modules/auth/getters_test.go:25]: Do we really need this struct, can't we just use a map instead

File: space-cli/cmd/modules/auth/getters_test.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+					API:  "/v1/config/projects/{project}/user-management/provider/{id}",
+					Type: "auth-providers",
+					Meta: map[string]string{"id": "local-admin", "project": "myproject"},
+					Spec: map[string]interface{}{"enabled": true, "secret": "hello"},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "test-2",
```
Comment (sharadregoti) [inline space-cli/cmd/modules/auth/getters_test.go:71]: same here

File: space-cli/cmd/modules/auth/getters_test.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+				mockSchema.On(m.method, m.args...).Return(m.paramsReturned...)
+			}
+
+			utils.Client = &mockSchema
+			got, err := GetAuthProviders(tt.args.project, tt.args.commandName, tt.args.params)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetAuthProviders() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(len(got), len(tt.want)) {
```
Comment (sharadregoti) [inline space-cli/cmd/modules/auth/getters_test.go:109]: If you want to compare 2 arrays use
https://yourbasic.org/golang/compare-slices/
use the first method

File: space-cli/cmd/utils/transport.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
+package utils
```
Comment (sharadregoti) [inline space-cli/cmd/utils/transport.go:1]: This file should be in a package called **transport**
which  will be  a folder under **utils**

File: space-cli/cmd/utils/transport.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+	mock.Mock
+}
+
+// Get gets spec object during test
+func (m *MocketAuthProviders) Get(method, url string, params map[string]string, vPtr interface{}) error {
+	c := m.Called(method, url, params, vPtr)
+	a, _ := json.Marshal(c[1])
+	_ = json.Unmarshal(a, vPtr)
+	return c.Error(0)
+}
```
Comment (sharadregoti) [inline space-cli/cmd/utils/transport.go:81]: @YourTechBud where should we write this these mock structs in space cli

File: space-cli/cmd/modules/auth/getters_test.go (hunk 5 above)
Comment (YourTechBud) [reply space-cli/cmd/modules/auth/getters_test.go:101]: Do the needful

**Comment by YourTechBud** [comment]
Also tests are failing

**Pull request merged**